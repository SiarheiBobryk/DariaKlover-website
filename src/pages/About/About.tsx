import * as React from 'react';
import { Helmet } from 'react-helmet';

import Typography from '@mui/material/Typography';
import Link from '@mui/material/Link';

// TODO: It would be great to add a pic here
function About() {
  return (
    <>
      <Helmet>
        <title>–û–±–æ –º–Ω–µ</title>
      </Helmet>
      <Typography variant="h4" component="h1" gutterBottom>
        –û–±–æ –º–Ω–µ
      </Typography>
      <Typography gutterBottom>–ü—Ä–∏–≤–µ—Ç! –Ø - –î–∞—Ä—å—è –ö–ª–æ–≤–µ—Ä.</Typography>
      <Typography gutterBottom>
        –ö–ª–æ–≤–µ—Ä - –º–æ–π –ø—Å–µ–≤–¥–æ–Ω–∏–º. –û—Ç –∞–Ω–≥–ª–∏–π—Å–∫–æ–≥–æ <Typography component="i">Ààkl…ô äv…ôr</Typography> - –∫–ª–µ–≤–µ—Ä üçÄ
      </Typography>
      <Typography gutterBottom>
        –ú–Ω–µ –æ—á–µ–Ω—å –Ω—Ä–∞–≤–∏—Ç—Å—è –∏–¥–µ—è –ø–µ—Ä–µ–¥–∞—á–∏ —Å–º—ã—Å–ª–æ–≤ —á–µ—Ä–µ–∑ –æ–±—Ä–∞–∑—ã. –í —ç—Ç–æ–º –µ—Å—Ç—å —á—Ç–æ-—Ç–æ —É–≤–ª–µ–∫–∞—Ç–µ–ª—å–Ω–æ–µ, –∏–Ω—Ç–µ—Ä–µ—Å–Ω–æ–µ –∏
        —Ç–∞–∏–Ω—Å—Ç–≤–µ–Ω–Ω–æ–µ. –£ –∫–ª–µ–≤–µ—Ä–∞ –µ—Å—Ç—å –º–Ω–æ–≥–æ –∑–Ω–∞—á–µ–Ω–∏–π. –î–ª—è –º–µ–Ω—è - —ç—Ç–æ –æ–ª–∏—Ü–µ—Ç–≤–æ—Ä–µ–Ω–∏–µ –±–∞–ª–∞–Ω—Å–∞ –∏ –≥–∞—Ä–º–æ–Ω–∏–∏. –ü–æ-–º–æ–µ–º—É, –∑–Ω–∞—á–µ–Ω–∏–µ
        —ç—Ç–æ–≥–æ —Å–∏–º–≤–æ–ª–∞ –æ—á–µ–Ω—å —Ö–æ—Ä–æ—à–æ –æ—Ç—Ä–∞–∂–∞–µ—Ç –º–µ–Ω—è –∏ —Ç–æ, –∫ –µ–º—É —è —Å—Ç—Ä–µ–º–ª—é—Å—å, –≤–æ —á—Ç–æ –≤–µ—Ä—é, —á–µ–º –∑–∞–Ω–∏–º–∞–ª–∞—Å—å –∏ –ø—Ä–æ–¥–æ–ª–∂–∞—é
        –∑–∞–Ω–∏–º–∞—Ç—å—Å—è.
      </Typography>
      <Typography gutterBottom>
        –í –ø—Ä–æ—à–ª–æ–º —è –≤—Ä–∞—á —Ç–µ—Ä–∞–ø–µ–≤—Ç, –≤—Ä–∞—á –æ–±—â–µ–π –ø—Ä–∞–∫—Ç–∏–∫–∏, –º–∞—Å—Å–∞–∂–∏—Å—Ç –∏ –∫–æ—Å–º–µ—Ç–∏–∫. –Ø –∑–Ω–∞—é –Ω–µ –ø–æ–Ω–∞—Å–ª—ã—à–∫–µ, —á—Ç–æ –¥–ª—è —Å–æ—Ö—Ä–∞–Ω–µ–Ω–∏—è
        –∑–¥–æ—Ä–æ–≤—å—è –Ω—É–∂–Ω–æ –Ω–µ —Ç–æ–ª—å–∫–æ –≤–ª–∞–¥–µ—Ç—å –Ω—É–∂–Ω–æ–π –∏–Ω—Ñ–æ—Ä–º–∞—Ü–∏–µ–π, –Ω–æ –∏ —É–º–µ—Ç—å —Ä–∞–∑—É–º–Ω–æ –µ–µ –ø—Ä–∏–º–µ–Ω–∏—Ç—å.
      </Typography>
      <Typography gutterBottom>
        –°–µ–π—á–∞—Å —è —É—á—É—Å—å –Ω–∞ health –∫–æ—É—á–∞. –°—É—Ç—å –º–æ–µ–π –ø—Ä–æ—Ñ–µ—Å—Å–∏–∏ - –ø–æ–º–æ—á—å —á–µ–ª–æ–≤–µ–∫—É –Ω–∞–π—Ç–∏ –±–∞–ª–∞–Ω—Å. –ë–∞–ª–∞–Ω—Å –≤ –µ–≥–æ —Ñ–∏–∑–∏—á–µ—Å–∫–æ–º,
        —ç–º–æ—Ü–∏–æ–Ω–∞–ª—å–Ω–æ–º, –º–µ–Ω—Ç–∞–ª—å–Ω–æ–º –∏ –¥—É—Ö–æ–≤–Ω–æ–º –∑–¥–æ—Ä–æ–≤—å–µ.
      </Typography>
      <Typography gutterBottom>
        –ú–æ—è —Ä–æ–ª—å –∫–∞–∫ health –∫–æ—É—á–∞ - –ø–æ–º–æ—á—å —á–µ–ª–æ–≤–µ–∫—É —É–≤–µ—Ä–µ–Ω–Ω–æ –ø—Ä–æ–π—Ç–∏ –µ–≥–æ –ø—É—Ç—å –∫ –æ–∑–¥–æ—Ä–æ–≤–ª–µ–Ω–∏—é, —Å–æ–∑–¥–∞–≤–∞—è –±–ª–∞–≥–æ–ø—Ä–∏—è—Ç–Ω—É—é
        –∞—Ç–º–æ—Å—Ñ–µ—Ä—É –¥–ª—è –∏–∑–º–µ–Ω–µ–Ω–∏–π, —Å–Ω–∞–±–∂–∞—è –Ω–µ–æ–±—Ö–æ–¥–∏–º—ã–º–∏ –∑–Ω–∞–Ω–∏—è–º–∏ –∏ –∏–Ω—Å—Ç—Ä—É–º–µ–Ω—Ç–∞–º–∏ –¥–ª—è –ø—Ä–∏–æ–±—Ä–µ—Ç–µ–Ω–∏—è, —Å–æ—Ö—Ä–∞–Ω–µ–Ω–∏—è –∏ —É–ø—Ä–∞–≤–ª–µ–Ω–∏—è
        —Å–≤–æ–∏–º –∑–¥–æ—Ä–æ–≤—å–µ–º.
      </Typography>
      <Typography gutterBottom>
        –ü–æ–∑–Ω–∞–∫–æ–º–∏—Ç—å—Å—è —Å–æ –º–Ω–æ–π –ø–æ–ª—É—á—à–µ –≤—ã –º–æ–∂–µ—Ç–µ —á–µ—Ä–µ–∑ –º–æ–π –ò–Ω—Å—Ç–∞–≥—Ä–∞–º{' '}
        <Link href="https://www.instagram.com/daria.klover">@daria.klover</Link> –∞–∫–∫–∞—É–Ω—Ç. –¢–∞–º —è –≤—Ä–µ–º—è –æ—Ç –≤—Ä–µ–º–µ–Ω–∏
        –≤—ã–∫–ª–∞–¥—ã–≤–∞—é —Å—Ç–æ—Ä–∏–∑ –∏ –ø–æ—Å—Ç—ã –æ —Å–µ–±–µ, —Å–≤–æ–∏—Ö –æ—Å–æ–∑–Ω–∞–Ω–∏—è—Ö –∏ –≤–∞–∂–Ω—ã—Ö –¥–ª—è –º–µ–Ω—è —Å–æ–±—ã—Ç–∏—è—Ö.
      </Typography>
      <Typography gutterBottom>–ê –¥–∞–Ω–Ω—ã–π —Å–∞–π—Ç - —ç—Ç–æ –ø–ª–æ—â–∞–¥–∫–∞ –¥–ª—è –ø—Ä–æ—Ñ–µ—Å—Å–∏–æ–Ω–∞–ª—å–Ω–æ–≥–æ —Ç–≤–æ—Ä—á–µ—Å—Ç–≤–∞.</Typography>
      <Typography gutterBottom>–î–æ–±—Ä–æ –ø–æ–∂–∞–ª–æ–≤–∞—Ç—å –≤ –≥–æ—Å—Ç–∏!</Typography>
    </>
  );
}

export default About;
