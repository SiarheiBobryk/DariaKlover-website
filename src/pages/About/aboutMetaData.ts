export interface AboutMetaData {
  title: string;
  heading: string;
  path: string;
}

const aboutMetaData: AboutMetaData = {
  title: 'Обо мне',
  heading: 'Обо мне',
  path: '/about',
};

export default aboutMetaData;
