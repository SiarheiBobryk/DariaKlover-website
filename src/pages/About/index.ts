export { default } from './About';
export { default as aboutMetaData } from './aboutMetaData';
