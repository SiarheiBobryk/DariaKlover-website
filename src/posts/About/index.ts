export { default } from './About';
export { default as aboutConstants } from './About';
