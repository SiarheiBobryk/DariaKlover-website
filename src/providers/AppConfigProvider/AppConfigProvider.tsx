import * as React from 'react';

import { PaletteMode } from '@mui/material';

export interface MetaTags {
  description: string;
  keywords: string;
  'msapplication-TileColor': string;
}

export interface CalendlyAppointment {
  id: string;
  active: boolean;
  newTab: boolean;
  label: string;
}

export interface Config extends MetaTags {
  fullName: string;
  title: string;
  url: string;
  language: string;
  colorModeDefault: PaletteMode;
  themeColor: {
    light: string;
    dark: string;
  };
  instagram: {
    id: string;
    href: string;
    label: string;
  };
  telegram: {
    id: string;
    href: string;
    label: string;
  };
  whatsapp: {
    id: string;
    href: string;
    label: string;
  };
  calendly: {
    id: string;
    href: string;
    label: string;
    appointments: Array<CalendlyAppointment>;
    conformation: boolean;
  };
  email: string;
  github: string;
}

export const config: Config = {
  fullName: 'Дарья Кловер',
  title: 'Дарья Кловер - Специалист и коуч по здоровью',
  url: 'https://dariaklover.health',
  description:
    // eslint-disable-next-line max-len
    'Образовательный сайт о здоровом образе жизни, инструментах и способах его достижения. Дарья Кловер - врач общей практики, косметик, блогер, зож-энтузиаст, будущий специалист и коуч по здоровью.',
  keywords: 'хелс коуч, health coach, здоровье, красота, питание, гармония, сопровождение, антиэйдж',
  language: 'ru',
  colorModeDefault: 'light', // NOTE: the color-mode is managed by `ColorModeProvider`
  themeColor: {
    light: '#fff',
    dark: '#212121',
  },
  instagram: {
    id: '@daria.klover',
    href: 'https://www.instagram.com/daria.klover',
    label: 'Инстаграм Дарьи',
  },
  telegram: {
    id: 'dariaklover',
    href: 'https://t.me/dariaklover',
    label: 'Телеграм Дарьи',
  },
  whatsapp: {
    // TODO: Setup the branded account id (payed feature) later
    id: '+46738540708',
    href: 'https://wa.me/+46738540708',
    label: 'WhatsApp Дарьи',
  },
  calendly: {
    id: 'dariaklover',
    href: 'https://calendly.com/dariaklover',
    label: 'Календарь Дарьи',
    appointments: [
      // NOTE: Due to limitations of a free account on Calendly, only one appointment type should be active
      { id: 'trial', active: true, newTab: true, label: 'Ознакомительная встреча' },
      { id: 'regular', active: false, newTab: true, label: 'Heath-коучинг сессия' },
    ],
    conformation: false,
  },
  'msapplication-TileColor': '#ffc40d',
  email: 'dariaklover@gmail.com',
  github: 'https://github.com/SiarheiBobryk/DariaKlover-website',
};

export const AppConfigContext = React.createContext<Config>(config);

export interface AppConfigProviderProps {
  children: React.ReactElement;
}

function AppConfigProvider(props: AppConfigProviderProps) {
  const { children } = props;
  return <AppConfigContext.Provider value={config}>{children}</AppConfigContext.Provider>;
}

export default AppConfigProvider;
