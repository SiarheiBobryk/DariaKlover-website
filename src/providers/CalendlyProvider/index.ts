export { default } from './CalendlyProvider';
export * from './CalendlyProvider';
