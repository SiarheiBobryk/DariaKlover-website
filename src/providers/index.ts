export { default as ColorModeProvider } from './ColorModeProvider';
export { default as ThemeProvider } from './ThemeProvider';
export { default as AppConfigProvider } from './AppConfigProvider';
