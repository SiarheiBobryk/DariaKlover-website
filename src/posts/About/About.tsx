import * as React from 'react';

import Typography from '@mui/material/Typography';
import Link from '@mui/material/Link';

import OpenInNewIcon from '@mui/icons-material/OpenInNew';
import { AppConfigContext, Config } from '../../providers/AppConfigProvider';
import Heading from '../../components/Heading';
import Paragraph from '../../components/Paragraph';

export const aboutConstants = {
  spell: `Ààkl…ô äv…ôr`,
};

export interface AboutProps {
  heading?: string;
}

function About(props: AboutProps) {
  const { heading } = props;
  const {
    instagram: { href: instagramHref, id: instagramId },
    fullName,
  }: Config = React.useContext(AppConfigContext);

  return (
    <>
      <Heading>{heading}</Heading>
      <Paragraph>–ü—Ä–∏–≤–µ—Ç! –Ø - {fullName}.</Paragraph>
      <Paragraph>
        –ö–ª–æ–≤–µ—Ä - –º–æ–π –ø—Å–µ–≤–¥–æ–Ω–∏–º. –û—Ç –∞–Ω–≥–ª–∏–π—Å–∫–æ–≥–æ <Typography component="i">{aboutConstants.spell}</Typography> - –∫–ª–µ–≤–µ—Ä üçÄ
      </Paragraph>
      <Paragraph>
        –ú–Ω–µ –æ—á–µ–Ω—å –Ω—Ä–∞–≤–∏—Ç—Å—è –∏–¥–µ—è –ø–µ—Ä–µ–¥–∞—á–∏ —Å–º—ã—Å–ª–æ–≤ —á–µ—Ä–µ–∑ –æ–±—Ä–∞–∑—ã. –í —ç—Ç–æ–º –µ—Å—Ç—å —á—Ç–æ-—Ç–æ —É–≤–ª–µ–∫–∞—Ç–µ–ª—å–Ω–æ–µ, –∏–Ω—Ç–µ—Ä–µ—Å–Ω–æ–µ –∏
        —Ç–∞–∏–Ω—Å—Ç–≤–µ–Ω–Ω–æ–µ. –£ –∫–ª–µ–≤–µ—Ä–∞ –µ—Å—Ç—å –º–Ω–æ–≥–æ –∑–Ω–∞—á–µ–Ω–∏–π. –î–ª—è –º–µ–Ω—è - —ç—Ç–æ –æ–ª–∏—Ü–µ—Ç–≤–æ—Ä–µ–Ω–∏–µ –±–∞–ª–∞–Ω—Å–∞ –∏ –≥–∞—Ä–º–æ–Ω–∏–∏. –ü–æ-–º–æ–µ–º—É, –∑–Ω–∞—á–µ–Ω–∏–µ
        —ç—Ç–æ–≥–æ —Å–∏–º–≤–æ–ª–∞ –æ—á–µ–Ω—å —Ö–æ—Ä–æ—à–æ –æ—Ç—Ä–∞–∂–∞–µ—Ç –º–µ–Ω—è –∏ —Ç–æ, –∫ –µ–º—É —è —Å—Ç—Ä–µ–º–ª—é—Å—å, –≤–æ —á—Ç–æ –≤–µ—Ä—é, —á–µ–º –∑–∞–Ω–∏–º–∞–ª–∞—Å—å –∏ –ø—Ä–æ–¥–æ–ª–∂–∞—é
        –∑–∞–Ω–∏–º–∞—Ç—å—Å—è.
      </Paragraph>
      <Paragraph>
        –í –ø—Ä–æ—à–ª–æ–º —è –≤—Ä–∞—á —Ç–µ—Ä–∞–ø–µ–≤—Ç, –≤—Ä–∞—á –æ–±—â–µ–π –ø—Ä–∞–∫—Ç–∏–∫–∏, –º–∞—Å—Å–∞–∂–∏—Å—Ç –∏ –∫–æ—Å–º–µ—Ç–∏–∫. –Ø –∑–Ω–∞—é –Ω–µ –ø–æ–Ω–∞—Å–ª—ã—à–∫–µ, —á—Ç–æ –¥–ª—è —Å–æ—Ö—Ä–∞–Ω–µ–Ω–∏—è
        –∑–¥–æ—Ä–æ–≤—å—è –Ω—É–∂–Ω–æ –Ω–µ —Ç–æ–ª—å–∫–æ –≤–ª–∞–¥–µ—Ç—å –Ω—É–∂–Ω–æ–π –∏–Ω—Ñ–æ—Ä–º–∞—Ü–∏–µ–π, –Ω–æ –∏ —É–º–µ—Ç—å —Ä–∞–∑—É–º–Ω–æ –µ–µ –ø—Ä–∏–º–µ–Ω–∏—Ç—å.
      </Paragraph>
      <Paragraph>
        –°–µ–π—á–∞—Å —è —É—á—É—Å—å –Ω–∞ health –∫–æ—É—á–∞. –°—É—Ç—å –º–æ–µ–π –ø—Ä–æ—Ñ–µ—Å—Å–∏–∏ - –ø–æ–º–æ—á—å —á–µ–ª–æ–≤–µ–∫—É –Ω–∞–π—Ç–∏ –±–∞–ª–∞–Ω—Å. –ë–∞–ª–∞–Ω—Å –≤ –µ–≥–æ —Ñ–∏–∑–∏—á–µ—Å–∫–æ–º,
        —ç–º–æ—Ü–∏–æ–Ω–∞–ª—å–Ω–æ–º, –º–µ–Ω—Ç–∞–ª—å–Ω–æ–º –∏ –¥—É—Ö–æ–≤–Ω–æ–º –∑–¥–æ—Ä–æ–≤—å–µ.
      </Paragraph>
      <Paragraph>
        –ú–æ—è —Ä–æ–ª—å –∫–∞–∫ health –∫–æ—É—á–∞ - –ø–æ–º–æ—á—å —á–µ–ª–æ–≤–µ–∫—É —É–≤–µ—Ä–µ–Ω–Ω–æ –ø—Ä–æ–π—Ç–∏ –µ–≥–æ –ø—É—Ç—å –∫ –æ–∑–¥–æ—Ä–æ–≤–ª–µ–Ω–∏—é, —Å–æ–∑–¥–∞–≤–∞—è –±–ª–∞–≥–æ–ø—Ä–∏—è—Ç–Ω—É—é
        –∞—Ç–º–æ—Å—Ñ–µ—Ä—É –¥–ª—è –∏–∑–º–µ–Ω–µ–Ω–∏–π, —Å–Ω–∞–±–∂–∞—è –Ω–µ–æ–±—Ö–æ–¥–∏–º—ã–º–∏ –∑–Ω–∞–Ω–∏—è–º–∏ –∏ –∏–Ω—Å—Ç—Ä—É–º–µ–Ω—Ç–∞–º–∏ –¥–ª—è –ø—Ä–∏–æ–±—Ä–µ—Ç–µ–Ω–∏—è, —Å–æ—Ö—Ä–∞–Ω–µ–Ω–∏—è –∏ —É–ø—Ä–∞–≤–ª–µ–Ω–∏—è
        —Å–≤–æ–∏–º –∑–¥–æ—Ä–æ–≤—å–µ–º.
      </Paragraph>
      <Paragraph>
        –ü–æ–∑–Ω–∞–∫–æ–º–∏—Ç—å—Å—è —Å–æ –º–Ω–æ–π –ø–æ–ª—É—á—à–µ –≤—ã –º–æ–∂–µ—Ç–µ —á–µ—Ä–µ–∑ –º–æ–π –ò–Ω—Å—Ç–∞–≥—Ä–∞–º{' '}
        <Link href={instagramHref} target="_blank">
          {instagramId} <OpenInNewIcon fontSize="small" sx={{ verticalAlign: 'middle' }} />
        </Link>
        . –¢–∞–º —è –≤—Ä–µ–º—è –æ—Ç –≤—Ä–µ–º–µ–Ω–∏ –≤—ã–∫–ª–∞–¥—ã–≤–∞—é —Å—Ç–æ—Ä–∏–∑ –∏ –ø–æ—Å—Ç—ã –æ —Å–µ–±–µ, —Å–≤–æ–∏—Ö –æ—Å–æ–∑–Ω–∞–Ω–∏—è—Ö –∏ –≤–∞–∂–Ω—ã—Ö –¥–ª—è –º–µ–Ω—è —Å–æ–±—ã—Ç–∏—è—Ö.
      </Paragraph>
      <Paragraph>–ê –¥–∞–Ω–Ω—ã–π —Å–∞–π—Ç - —ç—Ç–æ –ø–ª–æ—â–∞–¥–∫–∞ –¥–ª—è –ø—Ä–æ—Ñ–µ—Å—Å–∏–æ–Ω–∞–ª—å–Ω–æ–≥–æ —Ç–≤–æ—Ä—á–µ—Å—Ç–≤–∞.</Paragraph>
      <Paragraph>–î–æ–±—Ä–æ –ø–æ–∂–∞–ª–æ–≤–∞—Ç—å –≤ –≥–æ—Å—Ç–∏!</Paragraph>
    </>
  );
}

export default About;
