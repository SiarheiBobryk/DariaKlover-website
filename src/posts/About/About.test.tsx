import * as React from 'react';

import { render, RenderResult } from '@testing-library/react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
import { axe, toHaveNoViolations } from 'jest-axe';

import { typographyClasses } from '@mui/material/Typography';

import About, { aboutConstants } from './About';
import { config } from '../../providers/AppConfigProvider';

expect.extend(toHaveNoViolations);

describe('<About />', () => {
  describe('a11y', () => {
    it('should match a11y standards', async () => {
      const { container }: RenderResult = render(<About heading="Lorem Ipsulum" />);

      // TODO: Find out the right type here ðŸ¤”
      const results = await axe(container);
      expect(results).toHaveNoViolations();
    });
  });

  describe('snapshotting', () => {
    it('should render correctly without props', () => {
      const about: ReactTestRendererJSON | ReactTestRendererJSON[] | null = renderer.create(<About />).toJSON();

      expect(about).toMatchSnapshot();
    });

    it('should render correctly with props', () => {
      const about: ReactTestRendererJSON | ReactTestRendererJSON[] | null = renderer
        .create(<About heading="Lorem Ipsulum" />)
        .toJSON();

      expect(about).toMatchSnapshot();
    });
  });

  describe('Elements inspection', () => {
    it('should have the `h2` HTML element as heading', () => {
      render(<About />);

      const h2: HTMLHeadingElement | null = document.querySelector('h2');
      expect(h2).toBeInTheDocument();
    });

    it('should have the CSS classes from `h5` Typography variant by default', () => {
      render(<About />);

      const h2: HTMLHeadingElement | null = document.querySelector('h2');
      expect(h2).toHaveClass(typographyClasses.h5);
    });

    it('should have the right constants', () => {
      render(<About heading="Lorem Ipsulum" />);

      const h2: HTMLHeadingElement | null = document.querySelector('h2');
      expect(h2).toHaveTextContent('Lorem Ipsulum');

      const i: HTMLElement | null = document.querySelector('i');
      expect(i).toHaveTextContent(aboutConstants.spell);
    });

    it('should have the right `a` element with the right attributes', () => {
      render(<About />);

      const a: HTMLAnchorElement | null = document.querySelector('a');
      expect(a).toHaveTextContent(config.instagram.id);
      expect(a).toHaveAttribute('href', config.instagram.href);
      expect(a).toHaveAttribute('target', '_blank');
    });

    // TODO: find out how to check the space char between the link here and for other component ðŸ¤”
  });
});
