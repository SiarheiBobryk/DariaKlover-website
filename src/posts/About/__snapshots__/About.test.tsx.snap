// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> snapshotting should render correctly 1`] = `
Array [
  <h2
    className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
  >
    –û–±–æ –º–Ω–µ
  </h2>,
  <p
    className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    –ü—Ä–∏–≤–µ—Ç! –Ø - 
    –î–∞—Ä—å—è –ö–ª–æ–≤–µ—Ä
    .
  </p>,
  <p
    className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    –ö–ª–æ–≤–µ—Ä - –º–æ–π –ø—Å–µ–≤–¥–æ–Ω–∏–º. –û—Ç –∞–Ω–≥–ª–∏–π—Å–∫–æ–≥–æ 
    <i
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Ààkl…ô äv…ôr
    </i>
     - –∫–ª–µ–≤–µ—Ä üçÄ
  </p>,
  <p
    className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    –ú–Ω–µ –æ—á–µ–Ω—å –Ω—Ä–∞–≤–∏—Ç—Å—è –∏–¥–µ—è –ø–µ—Ä–µ–¥–∞—á–∏ —Å–º—ã—Å–ª–æ–≤ —á–µ—Ä–µ–∑ –æ–±—Ä–∞–∑—ã. –í —ç—Ç–æ–º –µ—Å—Ç—å —á—Ç–æ-—Ç–æ —É–≤–ª–µ–∫–∞—Ç–µ–ª—å–Ω–æ–µ, –∏–Ω—Ç–µ—Ä–µ—Å–Ω–æ–µ –∏ —Ç–∞–∏–Ω—Å—Ç–≤–µ–Ω–Ω–æ–µ. –£ –∫–ª–µ–≤–µ—Ä–∞ –µ—Å—Ç—å –º–Ω–æ–≥–æ –∑–Ω–∞—á–µ–Ω–∏–π. –î–ª—è –º–µ–Ω—è - —ç—Ç–æ –æ–ª–∏—Ü–µ—Ç–≤–æ—Ä–µ–Ω–∏–µ –±–∞–ª–∞–Ω—Å–∞ –∏ –≥–∞—Ä–º–æ–Ω–∏–∏. –ü–æ-–º–æ–µ–º—É, –∑–Ω–∞—á–µ–Ω–∏–µ —ç—Ç–æ–≥–æ —Å–∏–º–≤–æ–ª–∞ –æ—á–µ–Ω—å —Ö–æ—Ä–æ—à–æ –æ—Ç—Ä–∞–∂–∞–µ—Ç –º–µ–Ω—è –∏ —Ç–æ, –∫ –µ–º—É —è —Å—Ç—Ä–µ–º–ª—é—Å—å, –≤–æ —á—Ç–æ –≤–µ—Ä—é, —á–µ–º –∑–∞–Ω–∏–º–∞–ª–∞—Å—å –∏ –ø—Ä–æ–¥–æ–ª–∂–∞—é –∑–∞–Ω–∏–º–∞—Ç—å—Å—è.
  </p>,
  <p
    className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    –í –ø—Ä–æ—à–ª–æ–º —è –≤—Ä–∞—á —Ç–µ—Ä–∞–ø–µ–≤—Ç, –≤—Ä–∞—á –æ–±—â–µ–π –ø—Ä–∞–∫—Ç–∏–∫–∏, –º–∞—Å—Å–∞–∂–∏—Å—Ç –∏ –∫–æ—Å–º–µ—Ç–∏–∫. –Ø –∑–Ω–∞—é –Ω–µ –ø–æ–Ω–∞—Å–ª—ã—à–∫–µ, —á—Ç–æ –¥–ª—è —Å–æ—Ö—Ä–∞–Ω–µ–Ω–∏—è –∑–¥–æ—Ä–æ–≤—å—è –Ω—É–∂–Ω–æ –Ω–µ —Ç–æ–ª—å–∫–æ –≤–ª–∞–¥–µ—Ç—å –Ω—É–∂–Ω–æ–π –∏–Ω—Ñ–æ—Ä–º–∞—Ü–∏–µ–π, –Ω–æ –∏ —É–º–µ—Ç—å —Ä–∞–∑—É–º–Ω–æ –µ–µ –ø—Ä–∏–º–µ–Ω–∏—Ç—å.
  </p>,
  <p
    className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    –°–µ–π—á–∞—Å —è —É—á—É—Å—å –Ω–∞ health –∫–æ—É—á–∞. –°—É—Ç—å –º–æ–µ–π –ø—Ä–æ—Ñ–µ—Å—Å–∏–∏ - –ø–æ–º–æ—á—å —á–µ–ª–æ–≤–µ–∫—É –Ω–∞–π—Ç–∏ –±–∞–ª–∞–Ω—Å. –ë–∞–ª–∞–Ω—Å –≤ –µ–≥–æ —Ñ–∏–∑–∏—á–µ—Å–∫–æ–º, —ç–º–æ—Ü–∏–æ–Ω–∞–ª—å–Ω–æ–º, –º–µ–Ω—Ç–∞–ª—å–Ω–æ–º –∏ –¥—É—Ö–æ–≤–Ω–æ–º –∑–¥–æ—Ä–æ–≤—å–µ.
  </p>,
  <p
    className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    –ú–æ—è —Ä–æ–ª—å –∫–∞–∫ health –∫–æ—É—á–∞ - –ø–æ–º–æ—á—å —á–µ–ª–æ–≤–µ–∫—É —É–≤–µ—Ä–µ–Ω–Ω–æ –ø—Ä–æ–π—Ç–∏ –µ–≥–æ –ø—É—Ç—å –∫ –æ–∑–¥–æ—Ä–æ–≤–ª–µ–Ω–∏—é, —Å–æ–∑–¥–∞–≤–∞—è –±–ª–∞–≥–æ–ø—Ä–∏—è—Ç–Ω—É—é –∞—Ç–º–æ—Å—Ñ–µ—Ä—É –¥–ª—è –∏–∑–º–µ–Ω–µ–Ω–∏–π, —Å–Ω–∞–±–∂–∞—è –Ω–µ–æ–±—Ö–æ–¥–∏–º—ã–º–∏ –∑–Ω–∞–Ω–∏—è–º–∏ –∏ –∏–Ω—Å—Ç—Ä—É–º–µ–Ω—Ç–∞–º–∏ –¥–ª—è –ø—Ä–∏–æ–±—Ä–µ—Ç–µ–Ω–∏—è, —Å–æ—Ö—Ä–∞–Ω–µ–Ω–∏—è –∏ —É–ø—Ä–∞–≤–ª–µ–Ω–∏—è —Å–≤–æ–∏–º –∑–¥–æ—Ä–æ–≤—å–µ–º.
  </p>,
  <p
    className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    –ü–æ–∑–Ω–∞–∫–æ–º–∏—Ç—å—Å—è —Å–æ –º–Ω–æ–π –ø–æ–ª—É—á—à–µ –≤—ã –º–æ–∂–µ—Ç–µ —á–µ—Ä–µ–∑ –º–æ–π –ò–Ω—Å—Ç–∞–≥—Ä–∞–º
     
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
      href="https://www.instagram.com/daria.klover"
      onBlur={[Function]}
      onFocus={[Function]}
      target="_blank"
    >
      @daria.klover
       
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1xzgpfl-MuiSvgIcon-root"
        data-testid="OpenInNewIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
        />
      </svg>
    </a>
    . –¢–∞–º —è –≤—Ä–µ–º—è –æ—Ç –≤—Ä–µ–º–µ–Ω–∏ –≤—ã–∫–ª–∞–¥—ã–≤–∞—é —Å—Ç–æ—Ä–∏–∑ –∏ –ø–æ—Å—Ç—ã –æ —Å–µ–±–µ, —Å–≤–æ–∏—Ö –æ—Å–æ–∑–Ω–∞–Ω–∏—è—Ö –∏ –≤–∞–∂–Ω—ã—Ö –¥–ª—è –º–µ–Ω—è —Å–æ–±—ã—Ç–∏—è—Ö.
  </p>,
  <p
    className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    –ê –¥–∞–Ω–Ω—ã–π —Å–∞–π—Ç - —ç—Ç–æ –ø–ª–æ—â–∞–¥–∫–∞ –¥–ª—è –ø—Ä–æ—Ñ–µ—Å—Å–∏–æ–Ω–∞–ª—å–Ω–æ–≥–æ —Ç–≤–æ—Ä—á–µ—Å—Ç–≤–∞.
  </p>,
  <p
    className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    –î–æ–±—Ä–æ –ø–æ–∂–∞–ª–æ–≤–∞—Ç—å –≤ –≥–æ—Å—Ç–∏!
  </p>,
]
`;
